### PR TITLE
Feature/experiment collection

### LINKED ISSUE #259: `experiment_series` does not work for self-paced navigation of experiments
Toontjehoger allows to choose from a set of experiments, but these experiments are hard-coded. `experiment_series` is not generic enough yet to allow this (it is, literally, a series).

### LINKED ISSUE #606: Create Control Dashboard for Congo Tune Twins
When launching the Congo branch, a dashboard similar to https://toontjehoger.amsterdammusiclab.nl/toontjehoger should be the first screen the scientist sees. 

She should then have the choice to select entering data for 

~~1. Consent~~ -> done offline
~~2. Demographic Data~~ -> done offline
1. Visual example
2. Original Tune Twins
3. Spectral Tune Twins
4. Timing Tune Twins
5. Pitch Tune Twins

Selecting each choice takes the experimenter through one iteration of the game, then returns to the home screen.

### LINKED ISSUE #607: Create Analysis Dashboard for Congo Branch
After launching the Congo Branch of tune twins, the scientist should be able to see the current status of the data that has been collected. This should include:

* Number of participants successfully entered for each condition in #606 as summary table
* Confirmation the JSON file format is correct

### PR DESCRIPTION
**Pull request by BeritJanssen**
Close #606 and close #259 : instead of embedding an `experiment_series` within a rules file, this branch adds a dedicated frontend component (now called `ExperimentCollection`, open to alternative suggestions) and backend view which goes through the experiments in an `ExperimentSeries` (which I didn't rename just yet, as this is quite the migration hassle).

The reasoning is as follows: `first_experiments` and `last_experiments` will always be run in order. If an `ExpererimentSeries` has set `dashboard=True`, the `ExperimentCollection` component will show a dashboard of the `randomized_experiments`. Clicking on a tile will lead to that experiment.
Screenshot 2024-02-21 at 08 28 52

 otherwise, it will redirect to the next experiment of the `randomized_experiments`.

If an ExperimentCollection has a dashboard, the embedded experiments can run multiple times. In fact, right now, we wouldn't be able to exit the ExperimentCollection's dashboard phase and continue to `last_experiments`. This is functionality left for later. The scss (copied from Toontjehoger) also still needs some attention, but was kept mostly in its orginal form as a reminder to integrate other features of Toontjehoger (showing logos, scores, extra information) into this component as well.

NB: this is a breaking change. TODO after deploy: add slugs to the existing `ExperimentSeries` on acceptance and production.

### CODE REVIEW

**Review by drikusroor — COMMENTED** [review]
I've mainly looked at the frontend code as you requested @Evert-R for a review. The only thing that was bugging me a bit was the use of `useEffect` and `useState` in the `ExperimentCollection` component. I tried to test whether it would make a difference on my local machine but I can't seem to properly load the experiments because of some weird error (which I'll post as a separate comment) 😅.

**Comment by drikusroor** [comment]
https://github.com/Amsterdam-Music-Lab/MUSCLE/assets/8208970/358dad38-abda-4f06-a52c-7e4c39ba988d

Like I mentioned before: Apparently my experiments crash / freeze in some sort of infinite loop when navigating to them from the Experiment Collection dashboard. Perhaps not even related to this PR, I'll let you know whether this is the case or not.

```
Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.
Experiment@http://localhost:3000/static/js/bundle.js:1293:8
withRouter(Experiment)@http://localhost:3000/static/js/bundle.js:40433:36
Route@http://localhost:3000/static/js/bundle.js:40214:34
Switch@http://localhost:3000/static/js/bundle.js:40385:34
Router@http://localhost:3000/static/js/bundle.js:39896:34
BrowserRouter@http://localhost:3000/static/js/bundle.js:39553:40
App@http://localhost:3000/static/js/bundle.js:355:73
```

**Comment by drikusroor** [comment]
Design-wise I think it would be nice to leave some space above the dashboard panel on larger screens:

image

On smaller screens I think it would be nice to either
1. Leave no space between the panel and the edge of the viewport and remove the rounded border, or:
2. Add spacing between the panel and the viewport, both vertically and horizontally

image

File: frontend/src/components/ExperimentCollection/ExperimentCollection.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        if (!loadingExperimentCollection) {
+            if (experimentCollection) {
+                if (experimentCollection.dashboard) {
+                    setDashboard(experimentCollection.dashboard);
+                } else {
+                    setExperiment(experimentCollection);
+                }
+            }
+        }
+    }, [experimentCollection, loadingExperimentCollection, setDashboard])
```
[the remark this answers is not in the document]
Comment (BeritJanssen) [reply frontend/src/components/ExperimentCollection/ExperimentCollection.js:22]: I'll try, but to my understanding, we'd need the `useEffect` hook to update logic once the response from the request to the backend's `collection` view is in.

File: frontend/src/components/ExperimentCollection/ExperimentCollection.test.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        }
+        const experiment2 = {
+            slug: 'another_slug',
+            name: 'Another Experiment'
+        }
+        jest.mock('../../API', () => ({
+            useExperimentCollection: () => [false, {dashboard: [experiment1, experiment2]}]
+        }));
+        render(<ExperimentCollection match={{params: {slug: 'some_collection'}}}/>)
+    });
```
[the remark this answers is not in the document]
Comment (BeritJanssen) [reply frontend/src/components/ExperimentCollection/ExperimentCollection.test.js:32]: Yes, same problem as with `Experiment` tests right now: if I enable them, they'll go into an infinite loop. I'll see if I can fix it today with mocking `axios` over `API`.

File: frontend/src/components/ExperimentCollection/ExperimentCollection.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (BeritJanssen) [reply frontend/src/components/ExperimentCollection/ExperimentCollection.js:22]: Testing confirms your hunch that the hooks are unnecessary. Now that I think about it, `useGet` already wraps `useEffect`, so we'll get the rerender behaviour from there, of course.

**Comment by BeritJanssen** [comment]
As for the design, it is indeed a bit off now. But I'd rather not introduce spacing now, as the layout will change after #797. I also get the infinite loop error now! I'll have to look into why that happens, strange I didn't run into it before.

**Comment by BeritJanssen** [comment]
Fixed - the infinite loop was caused by a faulty merge, I think - a dependency was left in the `Experiment.useEffect` hook that shouldn't have been there.

**Review by Evert-R — APPROVED** [review]

**Pull request merged**